hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -242,11 +287,12 @@ class ExpressionPayload(record: GenericRecord,
    *
    * @return
    */
-  private def joinRecord(sourceRecord: IndexedRecord, targetRecord: IndexedRecord): IndexedRecord = {
+  private def joinRecord(sourceRecord: IndexedRecord, targetRecord: IndexedRecord): GenericRecord = {
```

Comment:
Taking `IndexedRecord` as input and returning `GenericRecord` as output. Why so? Is it because if we later indeed have to deserialize the record then we do so by name?

Reply:
It's just interface misalignment. Didn't want to overload this PR